http/server.go

```diff
@@ -212,7 +214,7 @@ func (h *handler) getRand(ctx context.Context, round uint64) ([]byte, error) {
 	}
 
 	// make sure we aren't going to ask for a round that doesn't exist yet.
-	if time.Unix(chain.TimeOfRound(h.chainInfo.Period, h.chainInfo.GenesisTime, round), 0).After(time.Now()) {
+	if time.Unix(chain.TimeOfRound(info.Period, info.GenesisTime, round), 0).After(time.Now()) {
```

Comment:
Calling `h.getChainInfo()` here is perhaps neater, but requires unnecessary locking.
Passing as a parameter encodes/enforces the current assumption that `getRand` is only called after a call to `h.getChainInfo()`